### PR TITLE
[RLlib] Deflake MARWIL and BC and remove memory leak from torch MARWIL policy

### PR DESCRIPTION
**Pull request by avnishn**
Here is the performance of MARWIL now. 
image

this is what bc looked like when I ran it a week ago....lets see what happens with the release test on ci
image

Its actually really fast when not running a ton of evaluation, so tbh we could move this to the CI.

memory leak was caused by a tensor which was having its operation added to the computation graph, but wasn't getting reaped from the graph since back prop never happened through it.
apparently this is pretty common:
https://discuss.pytorch.org/t/memory-leak-debugging-and-common-causes/67339

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
please do not merge unless these are passing:

https://buildkite.com/ray-project/release-tests-pr/builds/12314

File: release/rllib_tests/learning_tests/yaml_files/marwil/marwil-halfcheetahbulletenv-v0.yaml [hunk 1]
```diff
@@ -2,14 +2,18 @@ marwil-halfcheetahbulletenv-v0:
...
     stop:
         time_total_s: 3600
     config:
+        framework: "torch"
         # Use input produced by expert SAC algo.
-        input: ["~/halfcheetah_expert_sac.zip"]
+        input: "dataset"
+        input_config:
+            format: "json"
+            paths: "s3://air-example-data/rllib/half_cheetah/half_cheetah.json"
```
Comment (kouroshHakha) [inline release/rllib_tests/learning_tests/yaml_files/marwil/marwil-halfcheetahbulletenv-v0.yaml:16]: does s3 path work on release instances?

Comment (kouroshHakha) [reply release/rllib_tests/learning_tests/yaml_files/marwil/marwil-halfcheetahbulletenv-v0.yaml:16]: cuz it doesn't work for CI

**Review by kouroshHakha — APPROVED** [review]

File: release/rllib_tests/learning_tests/yaml_files/marwil/marwil-halfcheetahbulletenv-v0.yaml (hunk 1 above)
Comment (avnishn) [reply release/rllib_tests/learning_tests/yaml_files/marwil/marwil-halfcheetahbulletenv-v0.yaml:16]: yes it does

**Comment by avnishn** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/12452
merge pending this test passing

File: rllib/algorithms/marwil/marwil_torch_policy.py [hunk 2]
```diff
@@ -69,27 +69,32 @@ def loss(
...
 
             # Policy loss.
             # Update averaged advantage norm.
             rate = self.config["moving_average_sqd_adv_norm_update_rate"]
-            self._moving_average_sqd_adv_norm.add_(
-                rate * (adv_squared_mean - self._moving_average_sqd_adv_norm)
+            self._moving_average_sqd_adv_norm = (
+                rate * (adv_squared_mean.detach() - self._moving_average_sqd_adv_norm)
+                + self._moving_average_sqd_adv_norm
             )
```
Comment (gjoliver) [inline rllib/algorithms/marwil/marwil_torch_policy.py:81]: are we detaching everything now? :)

**Review by gjoliver — APPROVED** [review]
I would actually break this into 2 PRs.
one get rid of RB from marwil.
the other updates the release test config.
maybe let's keep PRs tackling one 1 thing/theme/problem next time.

**Comment by avnishn** [comment]
buildkite/ray-builders-pr/octopus-brain-tune-tests-and-examples-using-rllib — Failed
[Details](https://buildkite.com/ray-project/ray-builders-pr/builds/41354#01826545-1846-45de-8cef-812080694feb)

buildkite/ray-builders-pr/octopus-tune-tests-a-r-no-rllib — Failed
[Details](https://buildkite.com/ray-project/ray-builders-pr/builds/41354#01826545-17fb-4eb1-afb0-8c26c3151d86)


Failing tests seem unrelated. 
ml-oncall should potentially take a look. cc @krfricke

File: rllib/algorithms/marwil/marwil_torch_policy.py (hunk 2 above)
Comment (avnishn) [reply rllib/algorithms/marwil/marwil_torch_policy.py:81]: no detach 
https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html

returns a new tensor, its not an inplace op.

**Comment by avnishn** [comment]
> the other updates the release test config.
> maybe let's keep PRs tackling one 1 thing/theme/problem next time.

absolutely right. I was getting lazy here

**Comment by avnishn** [comment]
marwil passing:
https://buildkite.com/ray-project/release-tests-pr/builds/12432#01826556-b32e-4e9a-966a-14f2ccd8c1f6

bc passing:
https://buildkite.com/ray-project/release-tests-pr/builds/12314#01826148-e436-40ee-acfc-73452a52ee5a

**Comment by avnishn** [comment]
This is ready to merge
